sqlalchemy_exasol/merge.py

```diff
@@ -11,115 +11,136 @@
 
 
 class Merge(UpdateBase):
-
     __visit_name__ = "merge"
 
     def __init__(self, target_table, source_expr, on):
         self._target_table = target_table
         self._source_expr = source_expr
         self._on = on
-        self._on_columns = []
-        elements_to_check = list(on.get_children())
-        for e in elements_to_check:
-            if isinstance(e, Column):
-                if e.table == self._target_table:
-                    self._on_columns.append(e)
-            else:
-                elements_to_check.extend(e.get_children())
+        self._on_columns = list(
```

Comment:
I think, the logic doesn't equate the previous version, becaue the previous version looks like a serialized recursive descent into a tree